### PR TITLE
Experiment with Wear Compose pager

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Experiment with Wear Compose pager

#### WHY

See what upgrade to Compose 1.4 and Wear Compose 1.2 would be like. 

#### HOW

Replace Accompanist pager with Compose foundation.

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/pager/PagerScreen.kt [hunk 1]
```diff
@@ -94,12 +93,12 @@ public fun PagerScreen(
...
-public class PageScreenIndicatorState(private val state: PagerState) : PageIndicatorState {
+public class PageScreenIndicatorState(
+    private val state: PagerState,
     override val pageCount: Int
-        get() = state.pageCount
-
+) : PageIndicatorState {
     override val pageOffset: Float
-        get() = state.currentPageOffset
+        get() = state.currentPageOffsetFraction.takeIf { it.isFinite() } ?: 0f
```
Comment (yschimke) [inline compose-layout/src/main/java/com/google/android/horologist/compose/pager/PagerScreen.kt:101]: This resolved a crash.

File: compose-layout/src/androidTest/java/com/google/android/horologist/compose/pager/PagerScreenTest.kt [hunk 2]
```diff
@@ -80,7 +79,7 @@ class PagerScreenTest {
         }
 
         assertThat(state.currentPage).isEqualTo(0)
-        assertThat(state.pageCount).isEqualTo(5)
+//        assertThat(state.pageCount).isEqualTo(5)
```
Comment (BowerSteve) [inline compose-layout/src/androidTest/java/com/google/android/horologist/compose/pager/PagerScreenTest.kt:82]: Were you able to diagnose why this needs to be commented out?

Comment (yschimke) [reply compose-layout/src/androidTest/java/com/google/android/horologist/compose/pager/PagerScreenTest.kt:82]:
```
    internal val pageCount: Int
        get() = layoutInfo.totalItemsCount
```

**Comment by yschimke** [comment]
Premature, closing for now.

**Pull request closed without merging**